### PR TITLE
JW-1230 Refactor TestIntegration for Ansible

### PR DESCRIPTION
**Pull request by m-kostrzewa**
Split jobs to allow for 4-step pipeline: build, provision, deploy, test. 

Refactored some stuff (moved some files around, extracted common code).

### CODE REVIEW

**Comment by m-kostrzewa** [comment]
Retest this please

File: CIScripts/Deployment/AnsibleDeploy.ps1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+. $PSScriptRoot\..\Common.ps1
+
+function Deploy-Ansible {
+    Param ([Parameter(Mandatory = $true)] [int] $VMsNeeded)
+    $Job.Step("Deploying using Ansible", {
+        # TODO
```
Comment (m-kostrzewa) [inline CIScripts/Deployment/AnsibleDeploy.ps1]: @wurbanski @climebil TODO here

File: CIScripts/TestIntegrationCommon.ps1 [hunk 2]
```diff
@@ -1,126 +1,40 @@
...
+$TestbedSessions = $null
+$TestbedVMNames = $null
+
+if($Env:DEPLOY_METHOD -eq "Legacy") {
+    . $PSScriptRoot\Deployment\LegacyDeploy.ps1
+    $TestbedSessions, $TestbedVMNames = Deploy-Legacy -VMsNeeded 2 -IsReleaseMode $ReleaseModeBuild
+} elseif($Env:DEPLOY_METHOD -eq "Ansible") {
+    . $PSScriptRoot\Deployment\AnsibleDeploy.ps1
+    $IPs = Deploy-Ansible
+    # TODO ^^^^^^^^^^^^^^
```
Comment (m-kostrzewa) [inline CIScripts/TestIntegrationCommon.ps1]: @wurbanski @climebil TODO here

**Comment by m-kostrzewa** [comment]
Retest this please

File: CIScripts/TestIntegrationCommon.ps1 [hunk 3]
```diff
@@ -1,126 +1,41 @@
...
-        -DumpFilesLocation $DumpFilesLocation -DumpFilesBaseName $DumpFilesBaseName -MaxWaitVMMinutes $MaxWaitVMMinutes
+. $PSScriptRoot\RunTests.ps1
+. $PSScriptRoot\Job.ps1
+$Job = [Job]::new("Deploy and test")
+
+$ArtifactsDir = $Env:ARTIFACTS_DIR
+
+# TODO: for now.
+$Env:SHOULD_RUN_TESTS = $true
+$Env:DEPLOY_METHOD = "Legacy"
```
Comment (m-kostrzewa) [inline CIScripts/TestIntegrationCommon.ps1]: This will be removed when jenkins jobs are updated

**Comment by m-kostrzewa** [comment]
I launched pure tests from this and it passes (except the last command due to typo, but next commit fixes it ("Small fixes")) http://ci.wincontrail.codilime.com:8080/job/Wincontrail/job/mk-wip-new-testint/12/console

File: CIScripts/TestIntegrationCommon.ps1 [hunk 4]
```diff
@@ -1,126 +1,43 @@
...
-Test-DockerDriverMultiTenancy -Session $Sessions[0] -TestConfiguration $TestConfiguration
-Test-Pkt0PipeImplementation -Session $Sessions[0] -TestConfiguration $TestConfiguration
-
-if($Env:RUN_DRIVER_TESTS -eq "1") {
-    Test-DockerDriver -Session $Sessions[0] -TestConfiguration $TestConfiguration
+if($Env:DEPLOY_METHOD -eq "Legacy") {
+    . $PSScriptRoot\Deployment\LegacyDeploy.ps1
+    Teardown-Legacy -VMNames $TestbedVMNames
+} elseif($Env:DEPLOY_METHOD -eq "Ansible") {
+    # TODO
```
Comment (m-kostrzewa) [inline CIScripts/TestIntegrationCommon.ps1]: @wurbanski @climebil TODO fyi

File: CIScripts/TestIntegrationCommon.ps1 [hunk 5]
```diff
@@ -1,126 +1,45 @@
...
-$VMNames = [System.Collections.ArrayList] @()
-for ($i = 0; $i -lt $VMsNeeded; $i++) {
-    $VMNames += $VMBaseName + "-" + $i.ToString()
-}
+# TODO: remove when jenkins jobs are updated
+$Env:DEPLOY_METHOD = "Legacy"
 
-Write-Host "Starting Testbeds:"
-$VMNames.ForEach({ Write-Host $_ })
+$TestbedSessions = $null
```
Comment (kaweue) [inline CIScripts/TestIntegrationCommon.ps1]: Actually this initialization is not necessary.

File: CIScripts/TestIntegrationCommon.ps1 [hunk 6]
```diff
@@ -1,126 +1,45 @@
...
 
-if ($ReleaseModeBuild) {
-    $Sessions = New-TestbedVMs -VMNames $VMNames -InstallArtifacts $true -VIServerAccessData $VIServerAccessData `
-        -VMCreationSettings $VMCreationSettings -VMCredentials $VMCredentials -ArtifactsDir $ArtifactsDir `
-        -DumpFilesLocation $DumpFilesLocation -DumpFilesBaseName $DumpFilesBaseName -MaxWaitVMMinutes $MaxWaitVMMinutes
+if($Env:DEPLOY_METHOD -eq "Legacy") {
+    $TestbedSessions, $TestbedVMNames = Deploy-Legacy -VMsNeeded 2 -IsReleaseMode $ReleaseModeBuild
+} elseif($Env:DEPLOY_METHOD -eq "Ansible") {
+    $IPs = Deploy-Ansible
+    $TestbedSessions = New-RemoteSessions -VMNames $IPs -Credentials $Creds
```
Comment (kaweue) [inline CIScripts/TestIntegrationCommon.ps1]: $Creds is not set up.

File: CIScripts/RunTests.ps1 [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            GatewayIP = $Env:SNAT_GATEWAY_IP;
+            ContainerGatewayIP = $Env:SNAT_CONTAINER_GATEWAY_IP;
+            EndhostUsername = $Env:SNAT_ENDHOST_USERNAME;
+            EndhostPassword = $Env:SNAT_ENDHOST_PASSWORD;
+            DiskDir = $Env:SNAT_DISK_DIR;
+            DiskFileName = $Env:SNAT_DISK_FILE_NAME;
+            VMDir = $Env:SNAT_VM_DIR;
+        }
+
+        return #just for test
```
Comment (kaweue) [inline CIScripts/RunTests.ps1]: Just to block merge :) I please revert this patch.

File: CIScripts/TestIntegrationCommon.ps1 [hunk 8]
```diff
@@ -1,126 +1,45 @@
...
-    AdapterName = $Env:ADAPTER_NAME;
-    VMSwitchName = "Layered " + $Env:ADAPTER_NAME;
-    VHostName = "vEthernet (HNSTransparent)"
-    ForwardingExtensionName = $Env:FORWARDING_EXTENSION_NAME;
-    AgentConfigFilePath = "C:\ProgramData\Contrail\etc\contrail\contrail-vrouter-agent.conf";
-    DockerDriverConfiguration = $DockerDriverConfiguration;
+if($Env:SHOULD_RUN_TESTS) {
+    Run-Tests -Sessions $TestbedSessions
+} else {
+    Write-Output "Won't run tests."
```
Comment (kaweue) [inline CIScripts/TestIntegrationCommon.ps1]: Why not Write-Host ?

**Review by kaweue — CHANGES_REQUESTED** [review]

File: CIScripts/TestIntegrationCommon.ps1 [hunk 9]
```diff
@@ -1,126 +1,45 @@
...
 
-# Setting all variables needed for New-TestbedVMs from Environment
-. $PSScriptRoot\SetCommonVariablesForNewVMsFromEnv.ps1
+$ArtifactsDir = $Env:ARTIFACTS_DIR
 
-# There are always 2 VMs created for running all tests
-$VMsNeeded = 2
+# TODO: We will have a single script with multiple pipelines. This variable will be
+# set to different values in different pipelines (deploy-devenv vs test-integration)
+$Env:SHOULD_RUN_TESTS = $true
```
Comment (mclapinski) [inline CIScripts/TestIntegrationCommon.ps1]: Do you mean that this env will be set up by Jenkins? If yes then it's a very bad idea.

This shouldn't be an env var but a parameter set up by scripts and then passed forward.

**Review by mclapinski — DISMISSED** [review]

File: CIScripts/Common/Aliases.ps1 [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Create PSSessionT alias
+$AccelPSSessionT = [PowerShell].Assembly.GetType("System.Management.Automation.TypeAccelerators")
+$AccelPSSessionT::add("PSSessionT", "System.Management.Automation.Runspaces.PSSession")
+
+# Create PSCredentialT alias
+$AccelPSCredentialT = [PowerShell].Assembly.GetType("System.Management.Automation.TypeAccelerators")
+$AccelPSCredentialT::add("PSCredentialT", "System.Management.Automation.PSCredential")
```
Comment (kaweue) [inline CIScripts/Common/Aliases.ps1]: Missing new line at the end of file

File: CIScripts/TestIntegration.ps1 [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# Test-Integration job consolidates Provision, Deploy, Test and Teardown into
+# one job. It will be removed in the future, but for now we support it.
+
+. $PSScriptRoot\Provision.ps1
+. $PSScriptRoot\Deploy.ps1
+. $PSScriptRoot\Test.ps1
+. $PSScriptRoot\Teardown.ps1
```
Comment (kaweue) [inline CIScripts/TestIntegration.ps1]: New line at the end of line

**Review by kaweue — CHANGES_REQUESTED** [review]

**Review by jacek-izykowski — DISMISSED** [review]

**Comment by m-kostrzewa** [comment]
Retest this please

**Comment by m-kostrzewa** [comment]
Retest this please

**Review by kaweue — APPROVED** [review]

**Review by jacek-izykowski — APPROVED** [review]

**Pull request merged**